internal/olm/operator/registry/operator_installer.go

```diff
@@ -108,6 +109,11 @@ func (o OperatorInstaller) UpgradeOperator(ctx context.Context) (*v1alpha1.Clust
 		return nil, fmt.Errorf("error getting list of subscriptions: %v", err)
 	}
 
+	// If there are no subscriptions found, then the previous operator version doesn't exist, so return error
+	if len(subList.Items) == 0 {
+		return nil, errors.New("no existing operator found in the cluster to upgrade")
+	}
+
 	var subscription *v1alpha1.Subscription
```

Comment:
This could still be nil if no subscription exists where `s.Spec.Package == o.PackageName`.

Reply:
You're right. Thanks for catching that. 
https://github.com/operator-framework/operator-sdk/pull/4452